src/many-migration/src/lib.rs

```diff
@@ -362,117 +164,396 @@ impl<T, E> InnerMigration<'static, T, E> {
+    pub fn maybe_initialize_update_at_height(
+        &mut self,
+        storage: &mut T,
+        block_height: u64,
+    ) -> Result<(), E> {
+        if self.is_enabled() {
+            if block_height == self.metadata.block_height && !self.active {
+                self.active = true;
+                self.migration.initialize(storage)?;
+            } else if block_height > self.metadata.block_height {
+                self.migration.update(storage)?;
+            }
+        }
+
+        // Else ignore.
+        Ok(())
+    }
+
+    #[inline]
+    pub fn initialize(&self, storage: &mut T, block_height: u64) -> Result<(), E> {
+        if self.is_enabled() && block_height == self.metadata.block_height {
+            self.migration.initialize(storage)?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    pub fn update(&self, storage: &mut T, block_height: u64) -> Result<(), E> {
+        if self.is_enabled() && block_height > self.metadata.block_height {
+            self.migration.update(storage)?;
+        }
+        Ok(())
+    }
+
+    #[inline]
+    pub fn hotfix(&self, b: &[u8], block_height: u64) -> Option<Vec<u8>> {
+        if self.is_enabled() && self.metadata.block_height == block_height {
+            self.migration.hotfix(b)
+        } else {
+            None
+        }
+    }
+
+    #[inline]
+    pub fn is_regular(&self) -> bool {
+        matches!(self.migration.r#type, MigrationType::Regular(_))
+    }
+
+    #[inline]
+    pub fn is_hotfix(&self) -> bool {
+        matches!(self.migration.r#type, MigrationType::Hotfix(_))
+    }
+
+    #[inline]
+    pub fn name(&self) -> &str {
+        self.migration.name()
+    }
+
+    #[inline]
+    pub fn description(&self) -> &str {
+        self.migration.description()
+    }
+
+    #[inline]
+    pub fn metadata(&self) -> &Metadata {
+        &self.metadata
+    }
+
+    #[inline]
+    pub fn disable(&mut self) {
+        self.enabled = false;
+    }
+
+    #[inline]
+    pub fn enable(&mut self) {
+        self.enabled = true;
+    }
+
+    #[inline]
+    pub fn is_enabled(&self) -> bool {
+        self.enabled
+    }
+
+    #[inline]
+    pub fn is_active(&self) -> bool {
+        self.active
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct SingleMigrationConfig {
+    name: String,
 
     #[serde(flatten)]
     metadata: Metadata,
 }
 
-pub fn load_migrations<'a, 'b, E, T>(
-    registry: &'b [InnerMigration<'b, T, E>],
-    data: &'a str,
-) -> Result<BTreeMap<&'b str, Migration<'b, T, E>>, String> {
-    // TODO: Do not hardcode the deserializer
-    let config: Vec<MigrationJson> = serde_json::from_str(data).unwrap();
+impl<T, E> From<(&InnerMigration<T, E>, Metadata)> for SingleMigrationConfig {
+    fn from((migration, metadata): (&InnerMigration<T, E>, Metadata)) -> Self {
+        Self {
+            name: migration.name.to_string(),
+            metadata,
+        }
+    }
+}
+
+#[derive(Default, Serialize, Deserialize)]
```

Comment:
Please test serialization/deserialization of `MigrationConfig`

Reply:
really?

Reply:
Perhaps not. You seem to test it in fw anyway

Reply:
Yes, I agree